### DISCUSSION TITLE
#Beginner's guide

### DISCUSSION THREAD

**Discussion by RajivRanasingha** [DuajDiaz/Travel-Website#25] [Q&A] [answered]
Why String is final in Java?

**Comment by Bunny77K9** [comment, marked as answer]
The string is final because of the same reason it is immutable. A couple of reasons which I think make sense is an implementation of String pool, Security, and Performance. Java designers know that String will be used heavily in every single Java program, so they optimized it from the start.

**Reply by DuajDiaz** [reply]
Thank for your answer!
